rs/sns/audit/src/lib.rs

```diff
@@ -179,11 +179,7 @@ async fn validate_neurons_fund_sns_swap_participation(
     let refunded_neuron_portions = neurons_fund_refunds.neurons_fund_neuron_portions;
     let mut refunded_amounts_per_controller = BTreeMap::new();
     for refunded_neuron_portion in refunded_neuron_portions.iter() {
-        #[allow(deprecated)] // TODO(NNS1-3198): remove once hotkey_principal is removed
-        let controller = refunded_neuron_portion
-            .controller
-            .or(refunded_neuron_portion.hotkey_principal)
-            .unwrap();
+        let controller = refunded_neuron_portion.controller.unwrap();
```

Comment:
This won't work because we still need to audit old projects, that's kind of the point of being able to audit. :-) 

How about we keep the old version of the struct locally, try to decode the new one, if that fails — fall back to tryign to decode the old one? Does that make sense?

Reply:
With the migration this is no longer necessary